File: Algorithm-Swift/Algorithm-Swift/Inflearn/26_말아톤.swift
//
//  26_말아톤.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/06.
//

import Foundation

func solution26() {
    let n = Int(readLine()!)!
    let players = readLine()!.split(separator: " ").map { Int($0)! }
    var ranks = Array(repeating: 0, count: n)

    for i in 0..<n {
        var cnt = 0

        for j in 0..<i {
            if players[j] < players[i] {
                cnt += 1
            }
        }

        ranks[i] = (i + 1) - cnt
    }

    ranks.forEach {
        print($0, terminator: " ")
    }

}

File: Algorithm-CPP/26_말아톤.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int players[n], ranks[n];
    for(int i = 0; i < n; i++) {
        cin >> players[i];
    }

    for(int i = 0; i < n; i++) {
        int cnt = 0;

        for(int j = 0; j < i; j++) {
            if(players[j] < players[i]) cnt++;
        }

        ranks[i] = (i + 1) - cnt;
    }

    for(int i = 0; i < n; i++) {
        cout << ranks[i] << " ";
    }

    return 0;
}